pico/api/src/main/java/io/helidon/pico/api/InvocationContextBlueprint.java

```diff
@@ -52,21 +51,21 @@ public interface InvocationContext {
      *
      * @return the annotations on the enclosing type
      */
-    List<AnnotationAndValue> classAnnotations();
+    List<Annotation> classAnnotations();
 
     /**
      * The element info represents the method (or the constructor) being invoked.
      *
      * @return the element info represents the method (or the constructor) being invoked
      */
-    TypedElementName elementInfo();
+    TypedElementInfo elementInfo();
 
     /**
      * The method/element argument info.
      *
      * @return the method/element argument info
      */
-    Optional<TypedElementName[]> elementArgInfo();
+    List<TypedElementInfo> elementArgInfo();
```

Comment:
keeping it as an array was an optimization during execution.

Reply:
Working with both arrays and lists is a pain that is not worth the effort (except for highly optimized places in the code)